pbs_light.go

```diff
@@ -1034,8 +1053,15 @@ func NewFetchers(cfg *config.StoredRequests, db *sql.DB) (byId stored_requests.F
 
 	if cfg.InMemoryCache != nil {
 		glog.Infof("Using a Stored Request in-memory cache. Max size: %d bytes. TTL: %d seconds.", cfg.InMemoryCache.Size, cfg.InMemoryCache.TTL)
-		byId = stored_requests.WithCache(byId, in_memory.NewLRUCache(cfg.InMemoryCache))
-		byAmpId = stored_requests.WithCache(byAmpId, in_memory.NewLRUCache(cfg.InMemoryCache))
+		byIdCache := in_memory.NewLRUCache(cfg.InMemoryCache)
+		byAmpIdCache := in_memory.NewLRUCache(cfg.InMemoryCache)
+
+		byId = stored_requests.WithCache(byId, byIdCache)
+		byAmpId = stored_requests.WithCache(byAmpId, byAmpIdCache)
+
+		for _, ep := range eventProducers {
+			listeners = append(listeners, events.Listen(byIdCache, ep), events.Listen(byAmpIdCache, ep))
```

Comment:
This may not be desired as there could be overlaps between normal ids and AMP ids. There may need to be two separate `EventProducer`s with each cache listening to the appropriate one.